homeassistant/components/hausbus/config_flow.py

```diff
@@ -0,0 +1,113 @@
+"""Config flow for Haus-Bus integration."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any
+
+from pyhausbus.BusDataMessage import BusDataMessage
+from pyhausbus.de.hausbus.homeassistant.proxy.controller.data.ModuleId import ModuleId
+from pyhausbus.HomeServer import HomeServer
+from pyhausbus.IBusDataListener import IBusDataListener
+from pyhausbus.ObjectId import ObjectId
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_SCHEMA = vol.Schema({})
+
+
+class ConfigFlow(IBusDataListener, config_entries.ConfigFlow, domain=DOMAIN):  # type: ignore[misc]
+    """Handle a config flow for hausbus."""
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._found_device = False
+        self._search_task: asyncio.Task | None = None
+        self.home_server = HomeServer()
```

Comment:
Do we need to take any care to shut down the `HomeServer` object when ending the flow?

Reply:
This is a good point. For completeness we should remove the event listeners.

Reply:
I tried to add a method to remove the bus event listeners and call it from the search_complete step, but how do I call it in case the flow is aborted?

Reply:
You need to override this method https://github.com/home-assistant/core/blob/2462dfbc473232aa40ebcaf5861e0bedec6c256e/homeassistant/data_entry_flow.py#L881